docs/source/model_design/concept_model/standard_results_formats/index.rst

```diff
@@ -0,0 +1,197 @@
+
+Motivation behind standard formatting
+-------------------------------------
+
+Standardized formatting of Vivarium model results may be beneficial in the following ways:
+
+- Predictability of results format allows researchers to write results processing code in advance of receiving results
+
+- Results format consistency enables generalizable results processing code (for researchers and engineers) and reduces the need to code exception cases
+
+- Shared language and classification of terms may aid in communication between researchers and engineers
+
+Formatting guide
+-----------------
+
+General considerations
+++++++++++++++++++++++
+
+1. Consistency with GBD names is preferred over internal/other types of consistency. Generally, GBD names (such as age_group_name, location_name, cause_name, etc.) should appear in Vivarium results as :code:`{gbd_name}.replace(" ","_").lower()`. Note that sometimes GBD names things in inconsistent ways. For instance, "6-11 months" and "12 to 23 months". However, this will allow us to easily merge Vivarium results and GBD data using their shared names.
+
+2. Successful execution of these standard formats depends on researcher documentation of requested model outputs within the simulation concept model document.
+
+.. todo::
+
+  Link to updated template doc when ready
+
+3. If a researcher-requested output does not clearly fit within the examples outlined below, researchers and engineers should meet to determine desired results formatting and document the format within the project concept model requested outputs section.
+
+Parameters in results dataframes
+++++++++++++++++++++++++++++++++
+
+.. list-table:: Parameters
+  :header-rows: 1
+
+  * - Parameter
+    - Value type
+    - Definition
+    - Possible values
+  * - :code:`input_draw`
+    - integer
+    - Simulated input draw number
+    - [0, 999]
+  * - :code:`scenario`
+    - string
+    - Simulated scenario name
+    - * baseline
+      * alternative
+      * etc.
+  * - :code:`age`
+    - string
+    - Age group of stratum
+    - :code:`gbd_age_group_name.replace(" ","_").lower()`
+      
+      * early_neonatal
+      * 50_to_54
+      * etc.
+  * - :code:`sex`
+    - string
+    - Sex of stratum
+    - * male
+      * female
+  * - :code:`year`
+    - integer
+    - Simulated year
+    - * 2022 
+      * 2023
+      * etc.
+  * - :code:`{other_stratifying_variables}` (example: :code:`treatment`)
+    - string
+    - Some other variable by which to stratify values (in the same way as age, sex, year, etc.). Intervention coverage is a common example.
+    - Name and values of {other_stratifying_variables} to be defined in advance of implementation by researchers.
+
+
+      Example:
+
+      * covered
+      * uncovered
+  * - :code:`measure`
+    - string
+    - Description of what is being measured about the entity
+    - * ylds
+      * ylls
+      * deaths
+      * person_time
+      * transition_count
+      * first_moment
```

Comment:
Are these really measures? They feel like metrics. But, maybe it's fine to be inexact as long as we never want to use them at the same time as the other measures.

For the MM results CSVs, I had a column for the (equivalent GBD) metric. Seeing this makes me wonder if that could be more broadly useful/whether it should be a standard column here. 🤔

Reply:
GBD defines measures as deaths/dalys/ylds/etc. and metrics as rates/counts/etc., so I'm thinking that we're consistent in the terminology here? Otherwise I'm not sure I'm following this comment, maybe it's because I'm confused about measure/metric terms. Could you please provide an example of the column from the MM result CSVs?

Reply:
It wasn't that interesting (didn't vary), but here is an example of how I linked results to the Rate metric from GBD: https://github.com/ihmeuw/vivarium_research_multiple_myeloma/blob/main/results/china/Multiple%20Myeloma%20Output%20Shell%20Table%20-%20China.csv. The context was that they wanted it to be as similar to GBD outputs as possible.

I don't have much experience with the continuous exposures. My expectation was that the measure would be something like "continuous" and things like mean and SD would be metrics. That's just based on the fact that mean/SD/first-moment/second-moment/etc seem more like rate/count/etc than deaths/dalys/ylds.

I see GBD measure ID 19 is "Continuous" but there are no metrics that bear out my intuition.

All of this only really matters when we are merging with inputs. When you get e.g. mean and SD of a hemoglobin distribution, does it come with a measure and/or metric column? Or is it through an interface sufficiently different from `get_outputs` that there is no such concept?

Reply:
Ah, I see what you're saying. Continuous exposure mean and SDs typically come from get_draws as modelable entity IDs that don't really follow these naming conventions. So I think that GBD consistency is less critical here because of that, but still, maybe this is more intuitive? 

measure = continuous_exposure
entity = hemoglobin
sub_entities = first_moment, second_moment

Does that seem like a better fit?

Reply:
Yes, and I think it is more consistent with get_draws.

When you get_draws from the modelable entity for a continuous risk exposure, e.g. `get_draws(gbd_id_type="modelable_entity_id", gbd_id=10487, source='stgpr', location_id=6, release_id=6, year_id=2019)`, it comes back with measure_id 19 ("continuous") and metric_id 3 (for some reason, it is considered a rate?). This is also true when you use rei_id with source = "exposure" or "exposure_sd." So, there is some GBD precedent for treating different summary values like "sub-entities" as opposed to measures (or metrics, as it turns out).

Reply:
@NathanielBlairStahn here's some context for the first_moment classification, in case you hadn't seen it because it was hidden.

Reply:
> @NathanielBlairStahn point number 1 is [my fault](https://github.com/ihmeuw/vivarium_research/pull/1028#discussion_r1004919169). To me, "first moment" doesn't feel like a measure, and there is some consistency-with-GBD argument for it not being a measure, but if it seems like a measure to both of you, I'm happy to be overruled.

@zmbc @alibow I think it is less critical for Vivarium outputs to be consistent with GBD in the `measure` column than, for example, with cause names, risk names, age groups, or locations, as it is much easier to find the correct match for a measure (which has relatively few possible values in GBD) than for a cause, age group, etc. And as Ali said, GBD is not necessarily consistent about how it stores things that we might want to call a "measure," depending on where we pull the data from.

Also, I'm a bit confused by GBD's distinction between "measure" and "metric"; I think their goals with this distinction are different from our use cases in Vivarium outputs. For example, our outputs from Vivarium are (almost?) always "count space" values, so we should rarely if ever need to specify that an output is a "rate" rather than a "count" since our preferred methodology is to output the numerator and denominator separately. When I was originally thinking through which columns to put in a standardized output table, I considered having three separate columns, `measure`, `metric`, and `units`. After thinking through some examples, I was finding little utility for the `metric` column. And while `units` would be nice, oftentimes it is unnecessary (e.g., when the measure is DALYs, YLLs, or YLDs, which are already units), and otherwise we usually have conventions such as always measuring person-time in person-years. The exception would be something like hemoglobin concentration, which could be measured in g/L, g/dL, mmol/L, etc., so in these cases we'll just have to specify the desired units (consistently with GBD if possible) and remember what they are.

(to be continued...)

Reply:
So eventually I decided that the simplest thing to do would be to have a single `measure` column that would describe, as precisely as possible, the quantity being measured by the number we get in the output. The name of the measure should be some sort of physical quantity that can be measured concretely, and it can encompass both the "measure" and the "metric" in GBD's terminology (whatever those are supposed to mean). For example, if we wanted the count of deaths, the measure can just be called "deaths". If for some reason we wanted the simulation to directly output a mortality rate rather than a count, we could call the measure "death_rate". In my output processing code for the [CIFF SAM project](https://github.com/ihmeuw/vivarium_research_ciff_sam/tree/main/model_validation), I assigned new measures such as "prevalence", "transition_rate", or "rate_ratio" to derived quantities I computed from the raw count space outputs.

For a continuous quantity like hemoglobin concentration, it's a little more complicated; there are many possible aspects of the distribution of hemoglobin concentration in the population that we may want to measure. In previous simulations Vivarium has directly output the mean and standard deviation of the Hb concentration, in which case the measures could be called simply "mean" and "standard_deviation" (since those are the quantities that the numbers in the output are measuring). However, for consistency with other Vivarium outputs, and to make it easier to aggregate values over different population strata, I think it's better to output the numerators and denominators we need to compute the mean and standard deviation (if anyone disagrees, please chime in). In trying to describe what numerators we need to compute these (with person-time as the denominator), I came up with "first_moment" and "second_moment", based on the concept of "moment" from [physics](https://en.wikipedia.org/wiki/Moment_(physics)) and [mathematics](https://en.wikipedia.org/wiki/Moment_(mathematics)), so that's what I put in the `measure` column. If anyone has a better description for the quantities/measures defined by [Ali's equations](https://github.com/ihmeuw/vivarium_research/pull/1028/files#diff-27a94db806f9b410773a575674886b27c967e01728d7a689e49e74275fa34395R194-R196), I'm open to other suggestions. (Note the second moment can't be used directly to compute the variance or standard deviation as a ratio, but can be combined with the mean to compute the variance via the formula Var(Y) = E(Y^2) - (EY)^2.)

Reply:
We are definitely bikeshedding here; I'm fine with changing it back. I don't really _love_ either option, and only suggested the other one for more consistency with GBD. But, as we have discussed, with continuous exposures it requires a few steps to link with GBD anyway.

Reply:
Haha, yeah, all this discussion isn't really necessary to make this one decision, but I realized I had done a lot of thinking about the table design awhile ago, but it was all in my head, so I thought it might be good to document my reasoning and open it up for discussion in case anyone has ideas for improvement.